controllers/cluster_controller.go

```diff
@@ -205,6 +205,22 @@ func (r *ClusterReconciler) reconcileMetrics(_ context.Context, cluster *cluster
 func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster) (reconcile.Result, error) {
 	logger := r.Log.WithValues("cluster", cluster.Name, "namespace", cluster.Namespace)
 
+	// First handle the control plane
+	if cluster.Spec.ControlPlaneRef != nil {
+		obj, err := external.Get(ctx, r.Client, cluster.Spec.ControlPlaneRef, cluster.Namespace)
+		switch {
+		case apierrors.IsNotFound(err):
+		case err != nil:
+			return reconcile.Result{}, err
+		case obj.GetDeletionTimestamp().IsZero():
```

Comment:
@ncdc I don't have access to the struct fields here since it's an unstructured so I'm using the method call instead.